implementations/rust/ockam/ockam_transport_udp/src/workers/receiver.rs

```diff
@@ -69,17 +65,36 @@ impl Processor for UdpListenProcessor {
             }
         };
 
-        // Set return route to go directly to paired sender, skipping the UDP router
-        let new_route = route![
-            self.sender_addr.clone(),
-            Address::new(UDP, addr.to_string()),
-            msg.return_route(),
-        ];
-        msg = msg.set_return_route(new_route);
+        if msg.onward_route_ref().is_empty() {
+            return Ok(true);
+        }
+
+        let return_route = RouteBuilder::default().append(self.addresses.sender_address().clone());
```

Comment:
`route![self.addresses.sender_address().clone()]` should work there.

Reply:
Then you would need to convert it to `RouteBuilder`, which you can't because it can't reference a temporal value, so you'll need to create a new variable. I think it's better this way